packages/core/lib/commands/install.js

```diff
@@ -3,33 +3,43 @@ const command = {
   description: "Install a package from the Ethereum Package Registry",
   builder: {},
   help: {
-    usage: "truffle install <package_name>[@<version>]",
+    usage: "truffle install <packageId> [--alias]",
     options: [
       {
-        option: "package_name",
-        description:
-          "Name of the package as listed in the Ethereum Package Registry. (required)"
+        option: "packageId",
+        description: `(required) Name of the package as listed in the Ethereum Package Registry. 
+          Accepted formats: packageName, packageName@version, ethpm URI, ipfs URI.`
       },
       {
-        option: "<@version>",
+        option: "--alias",
         description:
-          "When specified, will install a specific version of the package, otherwise " +
-          "will install\n                    the latest version."
+          "A different name under which this package will be installed."
       }
     ]
   },
-  run: function(options, done) {
+  run: async function (options, done) {
     const Config = require("@truffle/config");
-    const Package = require("../package");
+    const PackageV1 = require("@truffle/ethpm-v1");
+    const PackageV3 = require("@truffle/ethpm-v3");
 
-    if (options._ && options._.length > 0) options.packages = options._;
+    if (options._ && options._.length == 0) {
+      done(new Error(`Please provide a packageId.`));
+    }
+    if (options._ && options._.length > 1) {
+      done(new Error(`Only one packageId can be installed at a time.`));
+    }
+    options.packageId = options._[0];
 
     const config = Config.detect(options);
-    Package.install(config)
-      .then(() => {
-        return done();
-      })
-      .catch(done);
+
+    if (config.ethpm.version == "1") {
```

Comment:
The ideal requirement I have in mind here is that `truffle install` will install whatever package, whether it's v1 or v3. Was this not possible? Are we forced to make the user decide between one or the other for installation? (I know that it's required to make this determination ahead of time for publish)

Reply:
Yeah, automatic version detection of manifests is not so straightforward, there are a lot of edge cases possible here. One thing to keep in mind is that the read api differs between the v1 registry and v3 registries. I feel we should encourage users to only use v3 (except for installing existing v1 packages). And in that case, they should be encouraged to update their packages to v3, and republish.

I guess this brings up another thought, which is adding a deprecation notice whenever users use v1. Do you have any objections to that?

Reply:
I don't have any problem with a deprecation notice. @gnidan?